addons/survey/tests/test_survey_invite.py

```diff
@@ -194,3 +194,26 @@ def test_survey_invite_token_internal(self):
             set(answers.mapped('email')),
             set([self.user_emp.email]))
         self.assertEqual(answers.mapped('partner_id'), self.user_emp.partner_id)
+
+    def test_survey_invite_token_by_email_nosignup(self):
+        Answer = self.env['survey.user_input']
+
+        self.env['res.partner'].create({
+            'name': 'Test 1',
+            'email': 'test@example.com',
+        })
+
+        self.env['res.partner'].create({
+            'name': 'Test 2',
+            'email': 'test@example.com',
+        })
+
+        self.survey.write({'access_mode': 'token', 'users_login_required': False})
+        action = self.survey.action_send_survey()
+        invite_form = Form(self.env[action['res_model']].with_context(action['context']))
+        invite_form.emails = 'test@example.com'
+        invite = invite_form.save()
+        invite.action_invite()
+
+        answers = Answer.search([('survey_id', '=', self.survey.id)])
+        self.assertEqual(len(answers), 1)
```

Comment:
We could also check that recipient is effectively the first found partner (based on display_name).